### PR TITLE
[NFC] Convert some fields on admin display preferences to use metdata

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
This has no functional impact - cleanup only.

As part of making settings forms more metadata-driven this is a further conversion

Affected field is bottom one in pic 
screenshot 2018-10-08 21 30 52

screenshot 2018-10-08 22 51 58

Before
----------------------------------------
Field is hard-coded

After
----------------------------------------
Field uses metadata

Technical Details
----------------------------------------
In general we want settings to be presented based on metadata & to transfer all those information assets into the xml - this is a step on that.

Comments
----------------------------------------

@mattwire

### CODE REVIEW

**Comment by JohnFF** [comment]
Reviewed and tested Admin Settings /civicrm/admin/setting/preferences/display?reset=1 
Doing others now.

File: templates/CRM/Admin/Form/Preferences/Display.tpl [hunk 1]
```diff
@@ -155,11 +155,11 @@
 
     <tr class="crm-preferences-display-form-block-preserve_activity_tab_filter">
       <td class="label"></td>
-      <td>{$form.preserve_activity_tab_filter.html} {$form.preserve_activity_tab_filter.label}</td>
+      <td>{$form.preserve_activity_tab_filter.html} </td>
     </tr>
     <tr class="crm-preferences-display-form-block-description">
       <td>&nbsp;</td>
-      <td class="description">{ts}When enabled, any filter settings a user selects on the contact's Activity tab will be remembered as they visit other contacts.{/ts}
+      <td class="description">{$settings_fields.preserve_activity_tab_filter.description}
```
Comment (JohnFF) [inline templates/CRM/Admin/Form/Preferences/Display.tpl]: I think the </td> should be on the line above.

**Review by JohnFF — CHANGES_REQUESTED** [review]

File: settings/Core.setting.php [hunk 2]
```diff
@@ -961,28 +966,28 @@
...
+    'html_type' => 'checkbox',
     'default' => '0',
     'add' => '4.7',
-    'title' => 'Preserve activity filters as a user preference',
+    'title' => ts('Preserve activity filters as a user preference'),
     'is_domain' => 1,
     'is_contact' => 0,
-    'description' => 'When enabled, any filter settings a user selects on the contact\'s Activity tab will be remembered as they visit other contacts',
-    'help_text' => NULL,
+    'description' => ts('When enabled, any filter settings a user selects on the contact\'s Activity tab will be remembered as they visit other contacts'),
```
Comment (JohnFF) [inline settings/Core.setting.php]: Please can we have a full stop at the end of the description?

File: settings/Core.setting.php [hunk 3]
```diff
@@ -961,28 +966,28 @@
...
     'group' => 'core',
     'name' => 'preserve_activity_tab_filter',
-    'type' => 'String',
-    'html_type' => 'Text',
+    'type' => 'Boolean',
+    'html_type' => 'checkbox',
     'default' => '0',
     'add' => '4.7',
-    'title' => 'Preserve activity filters as a user preference',
+    'title' => ts('Preserve activity filters as a user preference'),
```
Comment (JohnFF) [inline settings/Core.setting.php]: Title markup entries should probably end with a full stop like most of the description entries do.

File: CRM/Admin/Form/SettingTrait.php [hunk 4]
```diff
@@ -116,39 +116,42 @@ protected function addFieldsDefinedInSettingsMetadata() {
...
     foreach ($settingMetaData as $setting => $props) {
-      if (isset($props['quick_form_type'])) {
+      $quickFormType = $this->getQuickFormType($props);
+      if (isset($quickFormType)) {
+        $options = NULL;
         if (isset($props['pseudoconstant'])) {
           $options = civicrm_api3('Setting', 'getoptions', [
             'field' => $setting,
-          ]);
+          ])['values'];
```
[the remark this answers is not in the document]
Comment (eileenmcnaughton) [reply CRM/Admin/Form/SettingTrait.php]: I'm a big fan of that syntax though :-(

File: templates/CRM/Admin/Form/Preferences/Display.tpl [hunk 5]
```diff
@@ -155,11 +155,11 @@
 
     <tr class="crm-preferences-display-form-block-preserve_activity_tab_filter">
       <td class="label"></td>
-      <td>{$form.preserve_activity_tab_filter.html} {$form.preserve_activity_tab_filter.label}</td>
+      <td>{$form.preserve_activity_tab_filter.html} </td>
```
[the remark this answers is not in the document]
Comment (eileenmcnaughton) [reply templates/CRM/Admin/Form/Preferences/Display.tpl]: fixed

File: templates/CRM/Admin/Form/Preferences/Display.tpl (hunk 1 above)
Comment (eileenmcnaughton) [reply templates/CRM/Admin/Form/Preferences/Display.tpl]: fixed

File: settings/Core.setting.php (hunk 2 above)
Comment (eileenmcnaughton) [reply settings/Core.setting.php]: fixed

File: settings/Core.setting.php (hunk 3 above)
Comment (eileenmcnaughton) [reply settings/Core.setting.php]: as discussed - leaving full stops off the titles

**Comment by JohnFF** [comment]
Re-tested happy to merge :)

**Pull request merged**